src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -566,6 +566,9 @@ export class SearchView extends ViewPane {
 
 	private onSearchResultsChanged(event?: IChangeEvent): void {
 		if (this.isVisible()) {
+			if (this.editorService.activeEditor && this.editorService.activeEditor.shortTitle === event?.elements[0].name()) {
```

Comment:
For testing if the refresh comes from binding a notebook editor (which happens when we open it), I would rather see something like propagating whether the refresh is from calling `bindNotebookEditorWidget` in the searchModel to this function to test that. ie: adding something here to determine whether the refresh is from binding.

https://github.com/microsoft/vscode/blob/bdf6efedc54105a51f2cf9f012987e038625bfab/src/vs/workbench/contrib/search/browser/searchModel.ts#L855

Note that only changing that won't be sufficient because of the reasons I named in my other comment.